woeip/apps/air_quality/views.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import Error
+from django.shortcuts import redirect, render
+from django.utils.encoding import force_text
+
+from woeip.apps.air_quality import dustrak, forms, models
+
+logger = logging.getLogger(__name__)
+
+
+def index(request):
+    return render(request, 'index.html')
+
+
+@login_required
+def upload(request):
+    """Upload data for a session collected using the Dustrak air quality device and a separate GPS
+    log file. Creates a new Session instance, two SessionData instances, and Data instances for
```

Comment:
Nit: Consider ditching this second sentence. If your implementation changes, this docstring will probably not be updated, and grow stale. Your first sentence accurately describes what's going on, and is good enough.